Sprint 1/issue 16#redesign the events crud forms

DevLuis123 opened the pull request:
Added styles for CRUD parts, buttons, and textboxes

ErikGmz committed:
Initial issue no. 16 commit.

Co-authored-by: Alan Hernández <BlazeDex@users.noreply.github.com>
Co-authored-by: Loloxo <STR1ND3R@users.noreply.github.com>
Co-authored-by: Luis Gutierrez <DevLuis123@users.noreply.github.com>

DevLuis123 committed:
CRUD styles

Added styles for CRUD parts,buttons and textboxes

MauricioNC commented on app/views/home/index.html.erb:
```diff
@@ -9,5 +9,5 @@
 <h1>
-    Event managment system
+    Events management system
 </h1>
```
Add a blank line at the end of your code

ErikGmz replied:
Done!

MauricioNC commented on app/views/events/show.html.erb:
```diff
@@ -1,5 +1,12 @@
+method: :delete, data: {turbo_confirm: "Are you sure?"} %> 
+</div>
+<%= button_to 'Back', events_path, method: :get, class: 'btn-3' %>
+</div>
```
Add a blank line at the end of your code

ErikGmz replied:
Done!

MauricioNC commented on app/views/events/_event.html.erb:
```diff
@@ -1,7 +1,7 @@
-  <div style="margin-bottom: 5px"><b>Event location:</b> <%= event.location %></div>
-  <div style="margin-bottom: 15px"><b>Event price:</b> <%= event.price %></div>
+<div class= "grid grid-flow-row gap-3">
+  <div><b class="b">Event title:</b> <%= event.title%></div>
+  <div><b class="b">Event description:</b> <%= event.description %></div>
+  <div><b class="b">Event date:</b> <%= event.date %></div>
+  <div><b class="b">Event location:</b> <%= event.location %></div>
+  <div><b class="b">Event price:</b> <%= event.price %></div>
```
If you are using tailwind, why don't you apply the font styles directly in the <b></b> tags?

ErikGmz replied:
Fixed!

MauricioNC commented on app/views/events/new.html.erb:
```diff
@@ -2,4 +2,4 @@
   <%= render 'form', event: @event %>
 </div>
-<%= link_to 'Back', events_path %>
+<%= button_to 'Back', events_path, method: :get, class: 'btn-3 ml-5' %>
```
Use link_to for get requests, and button_to for delete and post requests

MauricioNC commented on app/views/events/edit.html.erb:
```diff
@@ -1,3 +1,3 @@
 <h1>Edit event</h1>
 <%= render 'form', event: @event %>
-<%= link_to 'Back', event_path(@event) %>
+<%= button_to 'Back', events_path, method: :get, class: 'btn-3 ml-5' %>
```
Use link_to for get requests, and button_to for delete and post requests

ErikGmz replied:
Done!

MauricioNC commented on app/views/events/_form.html.erb:
```diff
@@ -6,31 +6,36 @@
   </div> 
-    
-  <%= form.submit style: "margin-bottom: 15px" %> 
+  
+  
```
Remove extra empty lines

ErikGmz replied:
Fixed!

MauricioNC commented on test/events_controller_test.rb:
```diff
@@ -0,0 +1,37 @@
+    assert_response :success
+  end
+
+  test 'should create event' do
+    assert_difference('Event.count') do
+      post :create, params: { event: {
+                                        title: 'New Event',
+                                        description: 'Description',
+                                        date: '2024-01-25',
+                                        location: 'Location',
+                                        price: 10
+                                        }
+                                    }
+end
```
Indent correctly this code

ErikGmz replied:
This file was supposed to be in another branch, so it was deleted in this one and fixed in the other one.

juperameza-lizos reviewed (changes requested):
Take care of mauricio comments

ErikGmz committed:
link_to helper usage for GET requests, code
formatting fixes and complete Tailwind CSS
classes implementation into events views.

Co-authored-by: Alan Hernández <BlazeDex@users.noreply.github.com>
Co-authored-by: Loloxo <STR1ND3R@users.noreply.github.com>
Co-authored-by: Luis Gutierrez <DevLuis123@users.noreply.github.com>

juperameza reviewed (approved):
LGTM
Used my work account for the other comment lol
IF you could please delete that comment

MauricioNC reviewed (approved):
LGTM!
